[Examples] Add create-leo-app examples that execute credits.aleo functions (#1177)
AleoHQ/aleo: create-leo-app/template-react-credits-aleo-functions-ts/README.md

# React Credits.aleo Functions Template

This template demonstrates building execution transactions for all 6 credits.aleo

> transfer and record manipulation functions in a React application.

Reviewer: ```suggestion
This template demonstrates building execution transactions for 6 credits.aleo 
functions that demonstrate value transfers and record manipulation functions in a React application.
```

## Functions Demonstrated

### Public Transfers

> - **transfer_public**: Transfer credits between public balances

Reviewer: Generally code convention in Provable is to use periods in docs and code comments outside of headers. Add them to the text in this doc (minus the headers)

- **transfer_public_to_private**: Convert public credits to a private record

### Private Transfers

- **transfer_private**: Transfer credits using private records
- **transfer_private_to_public**: Convert private record to public balance

### Record Operations

- **join**: Combine two credit records into one
- **split**: Split one credit record into two

## Getting Started

```bash
npm install
npm run dev
```

Your app should be running on http://localhost:5173/

## Key Concepts

### Transaction Building

This template uses `buildExecutionTransaction` to create real transactions that
could be broadcast to the Aleo network. The SDK:

1. Generates zero-knowledge proofs locally in your browser
2. Fetches inclusion proofs from the network for private record operations

> 3. Builds complete transactions ready for broadcast

Reviewer: The inclusion proof is part of the overall zk proof generated!

```suggestion
2. Builds transactions from the proof ready for broadcast
```

**Note:** Transactions are built but not broadcast, so sample records remain
unspent and can be reused.

### Credits Records

> Private credits in Aleo are stored as records with this structure:

Reviewer: ```suggestion
Private credits in the credits.aleo program (the official representation of value on the Aleo Network) are stored as Aleo records with this structure:
```

```
{
  owner: aleo1xxx...xxx.private,
  microcredits: 1000000u64.private,
  _nonce: 123...789group.public,

>   _version: 1u8.public

Reviewer: ```suggestion
  owner: aleo1xxx...xxx.private, // An  aleo address that owns the record.
  microcredits: 1000000u64.private, // The amount in microcredits contained in the record.
  _nonce: 123...789group.public, // The unique nonce identifying the record.
  _version: 1u8.public // The version of the record.
```

}
```

### Microcredits

1 Aleo credit = 1,000,000 microcredits. All amounts are specified in
microcredits with the `u64` suffix (e.g., `50000u64`).

## Notes

> - Network connectivity is required for inclusion proofs

Reviewer: Network connectivity isn't actually always required.

```suggestion
- Network connectivity is required for this app
```

- Public function execution takes ~10-15 seconds

> - Private function execution takes ~30-120 seconds (includes network calls)

Reviewer: ```suggestion
- Private function execution takes ~30-120 seconds (Depending on whether keys are locally cached or not)
```
